tests/unit/restapi/v1/test_queue.py

```diff
@@ -0,0 +1,516 @@
+# This Software (Dioptra) is being made available as a public service by the
+# National Institute of Standards and Technology (NIST), an Agency of the United
+# States Department of Commerce. This software was developed in part by employees of
+# NIST and in part by NIST contractors. Copyright in portions of this software that
+# were developed by NIST contractors has been licensed or assigned to NIST. Pursuant
+# to Title 17 United States Code Section 105, works of NIST employees are not
+# subject to copyright protection in the United States. However, NIST may hold
+# international copyright in software created by its employees and domestic
+# copyright (or licensing rights) in portions of software that were assigned or
+# licensed to NIST. To the extent that NIST holds copyright in this software, it is
+# being made available under the Creative Commons Attribution 4.0 International
+# license (CC BY 4.0). The disclaimers of the CC BY 4.0 license apply to all parts
+# of the software developed or licensed by NIST.
+#
+# ACCESS THE FULL CC BY 4.0 LICENSE HERE:
+# https://creativecommons.org/licenses/by/4.0/legalcode
+"""Test suite for queue operations.
+
+This module contains a set of tests that validate the CRUD operations and additional
+functionalities for the queue entity. The tests ensure that the queues can be
+registered, renamed, deleted, and locked/unlocked as expected through the REST API.
+"""
+from __future__ import annotations
+
+from typing import Any
+
+import pytest
+from flask.testing import FlaskClient
+from flask_sqlalchemy import SQLAlchemy
+from werkzeug.test import TestResponse
+
+from dioptra.restapi.routes import V1_QUEUES_ROUTE, V1_ROOT
+
+from ..lib.actions import register_queue
+from ..lib.helpers import is_iso_format
+
+# -- Actions ---------------------------------------------------------------------------
+
+
+def rename_queue(
+    client: FlaskClient,
+    queue_id: int,
+    new_name: str,
+    new_description: str | None = None,
+) -> TestResponse:
+    """Rename a queue using the API.
+
+    Args:
+        client: The Flask test client.
+        queue_id: The id of the queue to rename.
+        new_name: The new name to assign to the queue.
+        new_description: The new description to assign to the queue.
+
+    Returns:
+        The response from the API.
+    """
+    payload = {"name": new_name}
+    if new_description:
+        payload["description"] = new_description
```

Comment:
Since this is a PUT, then a value for description must always be provided in the payload.

If `new_description` is None, then you'll need to issue a GET to retrieve the current value for description so you can include it in the PUT. Otherwise, the description will disappear, which isn't the expected behavior.